net/frr/Makefile

```diff
@@ -0,0 +1,413 @@
+  DEPENDS+=+frr-libfrr
+  TITLE:=STATICD routing engine
+endef
+
+define Package/frr-vtysh
+  $(call Package/frr/Default)
+  DEPENDS+=+frr-libfrr +libreadline +libncurses
+  TITLE:=integrated shell for frr routing software
+  CONFLICTS:=quagga-vtysh
+endef
+
+define Package/frr-watchfrr
+  $(call Package/frr/Default)
+  TITLE:=frr watchdog
+  DEPENDS+=+frr-libfrr
+  DEFAULT:=y if PACKAGE_frr
+endef
+
+define Package/frr-zebra
+  $(call Package/frr/Default)
+  TITLE:=Zebra daemon
+  DEPENDS+=+frr-libfrr
+  DEFAULT:=y if PACKAGE_frr
+  CONFLICTS:=quagga-zebra
+endef
+
+define Package/frr-libfrr/conffiles
+/etc/frr/
+endef
+
+#MAKE_PATH = ./build
+#CONFIGURE_PATH = ./build
+
+#it's still buggy
+HOST_CONFIGURE_ARGS += \
+    --enable-clippy-only="yes" \
+    --disable-ldpd \
+    --disable-bffd \
+    --disable-nhrpd \
+    --disable-watchfrr \
+    --disable-ospfapi \
+    --disable-ospfclient \
+    --disable-bgp-announce \
+    --disable-bgpd \
+    --disable-realms
+
+define Build/Prepare
+	$(Build/Prepare/Default)
+	mkdir -p $(PKG_BUILD_DIR)/build/lib
+	$(CP) $(PKG_BUILD_DIR)/lib/command.h $(PKG_BUILD_DIR)/build/lib/
+endef
+
+define Build/Configure
+    ( cd $(PKG_BUILD_DIR)/build/ ; \
+    cd build ; \
+    ../configure \
+	--host="$(GNU_TARGET_NAME)" \
+	--build="$(GNU_HOST_NAME)" \
+	CFLAGS="$(TARGET_CFLAGS) $(EXTRA_CFLAGS)" \
+	CXXFLAGS="$(TARGET_CFLAGS) $(EXTRA_CFLAGS)" \
+	LDFLAGS="$(TARGET_LDFLAGS) $(EXTRA_LDFLAGS)" \
+	HOST_CFLAGS="$(HOST_CFLAGS)" \
+	HOST_LDFLAGS="$(HOST_LDFLAGS)" \
+	BUILD_CPPFLAGS="$(TARGET_CPPLAGS)" \
+	BUILD_CFLAGS="$(TARGET_CFLAGS)" \
+	BUILD_LDFLAGS="$(TARGET_LDFLAGS)" \
+	--prefix=/usr \
+	--enable-shared \
+	--disable-static \
+	--enable-user=network \
+	--enable-group=network \
+	--enable-multipath=16 \
+	--disable-capabilities \
+	--disable-ospfclient \
+	--disable-doc \
+	--with-vtysh-pager=cat \
+	--localstatedir=/var/run/frr \
+	--sysconfdir=/etc/frr/ \
+	$(call autoconf_bool,CONFIG_PACKAGE_frr-babeld,babeld) \
+	$(call autoconf_bool,CONFIG_PACKAGE_frr-bfdd,bfdd) \
+	$(call autoconf_bool,CONFIG_PACKAGE_frr-bgpd,bgpd) \
+	$(call autoconf_bool,CONFIG_PACKAGE_frr-eigrpd,eigrpd) \
+	$(call autoconf_bool,CONFIG_PACKAGE_frr-fabricd,fabricd) \
+	$(call autoconf_bool,CONFIG_PACKAGE_frr-isisd,isisd) \
+	$(call autoconf_bool,CONFIG_PACKAGE_frr-ldpd,ldpd) \
+	$(call autoconf_bool,CONFIG_PACKAGE_frr-nhrpd,nhrpd) \
+	$(call autoconf_bool,CONFIG_PACKAGE_frr-ospfd,ospfd) \
+	$(call autoconf_bool,CONFIG_PACKAGE_frr-ospf6d,ospf6d) \
+	$(call autoconf_bool,CONFIG_PACKAGE_frr-pbrd,pbrd) \
+	$(call autoconf_bool,CONFIG_PACKAGE_frr-pimd,pimd) \
+	$(call autoconf_bool,CONFIG_PACKAGE_frr-staticd,staticd) \
+	$(call autoconf_bool,CONFIG_PACKAGE_frr-ripd,ripd) \
+	$(call autoconf_bool,CONFIG_PACKAGE_frr-ripngd,ripngd) \
+	$(call autoconf_bool,CONFIG_PACKAGE_frr-vtysh,vtysh) \
+	$(call autoconf_bool,CONFIG_PACKAGE_frr-libfrr,zebra) \
+)
+endef
+
+#	just speed it up
+NUM_CORES ?= $(shell grep -c "vendor_id" /proc/cpuinfo)
```

Comment:
Does this make sense? You have already set **PKG_BUILD_PARALLEL**?

Reply:
is not working when you use build like I'm using it here (we lose environment variables), and for now this is the only way

Reply:
Please explain. I have not seen this in other packages.

Reply:
the way `define Build/Configure` and `define Build/Compile` is defined, will make parallel not to work, you can test it.
in future I'll try to fix the clippy issue and this construct will not be needed anymore

Reply:
Overriding/trying to outsmart the toolchain is a no go and this would break if you were to run make -jX as far as I can tell.